File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while 1:
    n = int(input())
    
    if n == -1:
        break
    
    l = [i for i in range(1, n//2 + 1) if n % i == 0]
    
    if sum(l) == n:
        print(f'{n} = ', end='')
        for i in range(len(l) - 1):
            print(f'{l[i]} + ', end='')
        print(l[len(l) - 1])
    else:
        print(f'{n} is NOT perfect.')

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int n = sc.nextInt();
            if (n == -1) break; 

            List<Integer> divisors = new ArrayList<>();

            for (int i = 1; i <= n / 2; i++) {
                if (n % i == 0) {
                    divisors.add(i);
                }
            }

            int sum = 0;
            for (int d : divisors) {
                sum += d;
            }

            if (sum == n) {
                System.out.print(n + " = ");
                for (int i = 0; i < divisors.size(); i++) {
                    System.out.print(divisors.get(i));
                    if (i != divisors.size() - 1) System.out.print(" + ");
                }
                System.out.println();
            } else {
                System.out.println(n + " is NOT perfect.");
            }
        }

        sc.close();
    }
}
